go.mod

```diff
@@ -16,12 +16,12 @@ require (
 	go.uber.org/zap v1.26.0
 	golang.org/x/exp v0.0.0-20230817173708-d852ddb80c63
 	gopkg.in/yaml.v2 v2.4.0
-	k8s.io/api v0.27.7
-	k8s.io/apimachinery v0.27.7
-	k8s.io/client-go v0.27.7
-	k8s.io/component-base v0.27.7
+	k8s.io/api v0.28.5
+	k8s.io/apimachinery v0.28.5
+	k8s.io/client-go v0.28.5
+	k8s.io/component-base v0.28.5
```

Comment:
Since we are bumping these k8s dep versions do we also need to bump the KinD node image used for e2e testing to be based on 1.28?

Reply:
For a reference, I did this in https://github.com/operator-framework/catalogd/pull/227